crates/boundless-market/src/contracts/test_utils.rs

```diff
@@ -0,0 +1,316 @@
+//!
+//! NOTE: This module will be removed in later phases of the testnet and before mainnet.
+
+use crate::contracts::{
+    boundless_market::BoundlessMarketService,
+    bytecode::*,
+    hit_points::{default_allowance, HitPointsService},
+};
+use alloy::{
+    network::EthereumWallet,
+    node_bindings::AnvilInstance,
+    primitives::{Address, FixedBytes},
+    providers::{ext::AnvilApi, Provider, ProviderBuilder, WalletProvider},
+    signers::local::PrivateKeySigner,
+    sol_types::SolCall,
+};
+use alloy_primitives::B256;
+use anyhow::{Context, Ok, Result};
+use risc0_aggregation::SetInclusionReceiptVerifierParameters;
+use risc0_circuit_recursion::control_id::{ALLOWED_CONTROL_ROOT, BN254_IDENTITY_CONTROL_ID};
+use risc0_ethereum_contracts::set_verifier::SetVerifierService;
+use risc0_zkvm::{
+    sha::{Digest, Digestible},
+    Groth16ReceiptVerifierParameters,
+};
+
+pub struct TestCtx<P> {
+    pub verifier_addr: Address,
+    pub set_verifier_addr: Address,
+    pub hit_points_addr: Address,
+    pub boundless_market_addr: Address,
+    pub prover_signer: PrivateKeySigner,
+    pub customer_signer: PrivateKeySigner,
+    pub prover_provider: P,
+    pub prover_market: BoundlessMarketService<P>,
+    pub customer_provider: P,
+    pub customer_market: BoundlessMarketService<P>,
+    pub set_verifier: SetVerifierService<P>,
+    pub hit_points_service: HitPointsService<P>,
+}
+
+pub async fn deploy_verifier_router<P: Provider>(
+    deployer_provider: P,
+    owner: Address,
+) -> Result<Address> {
+    let instance = RiscZeroVerifierRouter::deploy(deployer_provider, owner)
+        .await
+        .context("failed to deploy RiscZeroVerifierRouter")?;
+    Ok(*instance.address())
+}
+
+pub async fn deploy_groth16_verifier<P: Provider>(
+    deployer_provider: P,
+    control_root: B256,
+    bn254_control_id: B256,
+) -> Result<Address> {
+    let instance =
+        RiscZeroGroth16Verifier::deploy(deployer_provider, control_root, bn254_control_id)
+            .await
+            .context("failed to deploy RiscZeroGroth16Verifier")?;
+    Ok(*instance.address())
+}
+
+pub async fn deploy_mock_verifier<P: Provider>(deployer_provider: P) -> Result<Address> {
+    let instance = RiscZeroMockVerifier::deploy(deployer_provider, FixedBytes::ZERO)
+        .await
+        .context("failed to deploy RiscZeroMockVerifier")?;
+    Ok(*instance.address())
+}
+
+pub async fn deploy_set_verifier<P: Provider>(
+    deployer_provider: P,
+    verifier_address: Address,
+    set_builder_id: Digest,
+) -> Result<Address> {
+    let instance = RiscZeroSetVerifier::deploy(
+        deployer_provider,
+        verifier_address,
+        <[u8; 32]>::from(set_builder_id).into(),
+        String::default(),
+    )
+    .await
+    .context("failed to deploy RiscZeroSetVerifier")?;
+    Ok(*instance.address())
+}
+
+pub async fn deploy_hit_points<P: Provider>(
+    deployer_signer: &PrivateKeySigner,
```

Comment:
Since you only need the address here, I would pass that in instead of the signer.
```suggestion
    owner_address: Address
```